src/notebook/Notebook.jl

```diff
@@ -204,13 +223,37 @@ function load_notebook_nobackup(io, path)::Notebook
             break
         else
             cell_id = UUID(cell_id_str)
-            code_raw = String(readuntil(io, _cell_id_delimiter))
+            
+            metadata_toml_lines = String[]
+            initial_code_line = ""
+            while !eof(io)
+                line = String(readline(io))
+                if startswith(line, _cell_metadata_prefix)
+                    prefix_length = ncodeunits(_cell_metadata_prefix)
+                    push!(metadata_toml_lines, line[begin+prefix_length:end])
+                else
+                    initial_code_line = line
+                    break
+                end
+            end
+
+            code_raw = initial_code_line * "\n" * String(readuntil(io, _cell_id_delimiter))
             # change Windows line endings to Linux
             code_normalised = replace(code_raw, "\r\n" => "\n")
+
+            # remove the disabled on startup comments for further processing in Julia
+            code_normalised = replace(replace(code_normalised, _disabled_prefix => ""), _disabled_suffix => "")
+
             # remove the cell suffix
             code = code_normalised[1:prevind(code_normalised, end, length(_cell_suffix))]
 
-            read_cell = Cell(cell_id, code)
+            # parse metadata
```

Comment:
Give answers to "why?" in comments and not answers to "what?". "what?" is already answered by the code. If a comment is used to denote a block of lines doing some bigger task, then consider moving it into a function. Julia inlines functions anyway so during runtime they are mostly free

Reply:
While I agree that separating code into smaller functions is in general a good style, it would be different to the style of the existing code in this file. In addition, the code in this block has multiple output variables, making it harder to read than just having the code in a block. And for code blocks, I think a short high-level statement what the block does (like section headers in a book) is helpful for orientation.

Reply:
I think you are both right :)